net/http/inc/THttpLongPollEngine.h

```diff
@@ -0,0 +1,43 @@
+// $Id$
+// Author: Sergey Linev   8/01/2018
+
+/*************************************************************************
+ * Copyright (C) 1995-2017, Rene Brun and Fons Rademakers.               *
+ * All rights reserved.                                                  *
+ *                                                                       *
+ * For the licensing terms see $ROOTSYS/LICENSE.                         *
+ * For the list of contributors see $ROOTSYS/README/CREDITS.             *
+ *************************************************************************/
+
+#ifndef ROOT_THttpLongPollEngine
+#define ROOT_THttpLongPollEngine
+
+#include "THttpWSEngine.h"
+
+#include <string>
+#include <list>
+
+class THttpLongPollEngine : public THttpWSEngine {
+protected:
+   THttpCallArg *fPoll;              ///!< polling request, which can be used for the next sending
+   std::list<std::string> fBuf;      ///!< entries submitted to client
```

Comment:
`list` is almost always worse than `vector`, especially because resizing is so cheap these days (moved strings). So unless you need to the element reshuffling capabilities ("remove the second-to-last element") you're much better off with `vector`. (Same for other code here.)

Reply:
fBuf is wrong name here - it is queue. And I need to remove elements in the front - `pop_front()`. There is no equivalent in the vector